container-runtime/podman_config.sh

```diff
@@ -0,0 +1,117 @@
+HOME_CONTAINERS_CONF=~/.config/containers/containers.conf
+ETC_CONTAINERS_CONF=/etc/containers/containers.conf
+DOCKER_INIT=/usr/libexec/docker/docker-init
+KRUN_PATH=/opt/kontain/bin/krun
+KM_PATH=/opt/kontain/bin/km
+KM_SELINUX_CONTEXT="system_u:object_r:bin_t:s0"
+
+# Either do it or just print the commands we think should be run.
+#DEBUG=echo
+DEBUG=""
+
+# Install podman
+if ! dnf list installed | grep -q podman
+then
+   echo "podman is not installed, installing it"
+   $DEBUG sudo dnf -y install podman
+else
+   echo "podman is installed"
+fi
+
+
+# set init_path in the [containers] section of ~/.config/containers/containers.conf
+# We don't ensure we add this in the [containers] section.
+# We hope the init_path command line exists and is in the proper section.
+echo
+if ! grep -q "^ *init_path" $HOME_CONTAINERS_CONF
+then
+   # no init_path, use the docker init
+   $DEBUG sed -e "/^# *init_path/ainit_path = \"$DOCKER_INIT\"" $HOME_CONTAINERS_CONF >/tmp/sed$$.out
+   if test $? -eq 0
+   then
+      mv /tmp/sed$$.out $HOME_CONTAINERS_CONF
+   fi
+   rm -f /tmp/sed$$.out
+else
+   if ! grep -q "^ *init_path *= *\"$DOCKER_INIT\"" $HOME_CONTAINERS_CONF
+   then
+      echo "Leaving init_path statement in $HOME_CONTAINERS_CONF unchanged, you may need to change it to $DOCKER_INIT"
+      echo "Existing init_path:"
+      grep "^ *init_path" $HOME_CONTAINERS_CONF
+   else
+      echo "init_path = \"$DOCKER_INIT\"" is already in $HOME_CONTAINERS_CONF
+   fi
+fi
+
+
+# add krun to ~/.config/containers/containers.conf at the beginning of the [engine.runtimes] section
+echo
+if ! grep -q "^ *krun =" $HOME_CONTAINERS_CONF
+then
+   $DEBUG sed -e "/^\[engine.runtimes\]/akrun = [\n   \"$KRUN_PATH\",\n]\n" $HOME_CONTAINERS_CONF >/tmp/sed$$.out
+   if test $? -eq 0
+   then
+      mv /tmp/sed$$.out $HOME_CONTAINERS_CONF
+   fi
+   rm -f /tmp/sed$$.out
+else
+   echo "runtime krun already configured in $HOME_CONTAINERS_CONF"
+   grep -A 3 "^ *krun =" $HOME_CONTAINERS_CONF
+fi
+
+
+# set selinux context on /opt/kontain/bin/km
+echo
+$DEBUG chcon $KM_SELINUX_CONTEXT $KM_PATH
+
+
+# Get policy build tools.
+echo
+if ! dnf list installed | grep -q selinux-policy-devel
+then
+   echo "selinux-policy-devel is not installed, installing it"
+   $DEBUG sudo dnf -y install selinux-policy-devel
+else
+   echo "selinux-policy-devel is already installed"
+fi
+
+
+# Add kontain selinux policy adjustments
+POLDIR=/tmp/kontain_selinux_policy
+if ! test -d $POLDIR
+then
+   mkdir $POLDIR
+fi
+pushd $POLDIR
+cat <<EOF >kontain_selinux_policy.te
+module kontain_selinux_policy 1.0.0;
+
+require {
+  type container_t;
+  type kvm_device_t;
+  class chr_file { append getattr ioctl lock open read write };
+}
+
+allow container_t kvm_device_t:chr_file { append getattr ioctl lock open read write };
+
+# You may get errors when this is "compiled".
+# This is a known issue, see:
+# https://bugzilla.redhat.com/show_bug.cgi?id=1861968
+EOF
+
+if ! test -e Makefile
```

Comment:
Can the Makefile come from anywhere else? If not `ln -sf` would just do the same in one line